src/firefly/js/visualize/saga/CoverageWatcher.js

```diff
@@ -212,49 +212,56 @@ function removeCoverage(tbl_id, decimatedTables) {
 function updateCoverage(useHiPS, tbl_id, viewerId, decimatedTables, options) {
 
     if (!tbl_id) return null;
-    const table= getTblById(tbl_id);
-    if (!table) return null;
-    if (!options.hasCoverageData(options, table)) return null;
-    if (decimatedTables[tbl_id]==='WORKING') return tbl_id;
-
-
-    const params= {
-        startIdx : 0,
-        pageSize : MAX_ROW,
-        inclCols : getCovColumnsForQuery(options, table)
-    };
-
-    let req = cloneRequest(table.request, params);
-    if (table.totalRows>10000) {
-        const cenCol= options.getCenterColumns(table);
-        params.decimate=  serializeDecimateInfo(cenCol.lonCol, cenCol.latCol, 10000);
-        req = makeTableFunctionRequest(table.request, 'DecimateTable', 'coverage',
-            {decimate: serializeDecimateInfo(cenCol.lonCol, cenCol.latCol, 10000), pageSize: MAX_ROW});
-    }
+    try {
+        const table = getTblById(tbl_id);
+        if (!table) return null;
+        if (!options.hasCoverageData(options, table)) return null;
```

Comment:
minor thing, instead of passing itself to the function defined in 'options', I guess you could access the property from 'options' by using 'this'.